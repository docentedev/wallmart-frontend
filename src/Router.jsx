import React from 'react'
import {
    BrowserRouter,
    Switch,
    Route,
} from 'react-router-dom'
import Cart from './containers/cart/Cart'
import Home from './containers/home/Home'
import Header from './components/header/Header'

function Router({ products }) {
    return (
        <BrowserRouter>
            <Header />
            <Switch>
                <Route path="/cart">
                    <Cart />
                </Route>
                <Route path="/">
                    <Home products={products} />
                </Route>
            </Switch>
        </BrowserRouter>
    );
}

export default Router;
