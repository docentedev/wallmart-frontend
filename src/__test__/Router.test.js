import React from 'react'
import TestRenderer from 'react-test-renderer'

import Router from '../Router'

describe('Router test', () => {
    it('should render Router component', () => {
        const testRenderer = TestRenderer.create(<Router />)
        expect(testRenderer.toJSON()).toMatchSnapshot()
    })
})
