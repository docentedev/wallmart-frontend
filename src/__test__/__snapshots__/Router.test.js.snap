// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router test should render Router component 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="header__logo_container"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Lider Logo"
          src="lider-logo.svg"
        />
      </a>
    </div>
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        Productos
      </a>
      <a
        href="/cart"
        onClick={[Function]}
      >
        Carrito
      </a>
    </div>
  </header>,
  <div>
    <section
      className="area_products"
    >
      <div>
        Cargando...
      </div>
    </section>
  </div>,
]
`;
